src/Appwrite/Platform/Workers/Builds.php

```diff
@@ -456,6 +451,10 @@ protected function buildDeployment(callable $getFunctionsDevice, Func $queueForF
                 ]);
 
             if ($err) {
+                $build = $dbForProject->getDocument('builds', $buildId);
+                if ($build->getAttribute('status') === 'cancelled') {
+                    return;
```

Comment:
Here: https://github.com/appwrite/appwrite/pull/7605/files#diff-e06b561b3950351ec356b5f1b2ecc67b6999336f9efb3cb339fae4a67797b4e0L497

I would also ass one more cancelled check - so it doesn't get overriten with "failed"